source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -300,7 +311,10 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
     const std::string new_url = fmt::format(
         AuthorizationEndpointFormat, config_->authorizationEndpoint(), config_->clientId(),
         config_->encodedAuthScopes(), escaped_redirect_uri, escaped_state);
-    response_headers->setLocation(new_url);
+
+    const std::string& resource_param = config_->encodedResources();
+
+    response_headers->setLocation(new_url + resource_param);
```

Comment:
Just a small nit but could we rename `encodedResources` to `encodedResourceQueryParams` and then inline `resource_param` like so:

```
response_headers->setLocation(new_url + config_->encodedResourceQueryParams());
```

? I think that would read a bit cleaner

Reply:
@snowp, I have done the changes. The macos tests are failing because go binary could not be installed on the test setup. 
Let me know how to move forward from here.

code is up for review